mantidproject/mantid
# Loading scan mode data for IN5, PANTHER, SHARP

This PR implements the new data taking workflow recently added to IN5 and PANTHER instruments at the ILL. The scan data structure is the same as in scanning diffraction instruments at the ILL, thus the scan mode for IN5 and PANTHER is loaded through LoadILLDiffraction, with minor changes, to avoid code duplication.

This PR also fixes a recent issue (#30780) caused by scan data type change, that resulted in segmentation violations when an attempt was made to load data from the current data-taking cycle. This was caused by the type change from "ILL_detectors_data_scan" to "NXill_detectors_data_scan" .

The LoadILLTest has been filled with missing tests for D1B, and D7, as well as the new mode for IN5.

**To test:**

1. Get the new unit test data for IN5 scan mode: ILL/IN5/199857.nxs
2. Try to load data, check the instrument viewer, check if the sample logs are filled, e.g. wavelength

To test the solution of #30780 , if you access to ILL data:
1. Try to load internalUse data from the cycle 211.
2. There should be no issues, the sampleLogs should be properly filles.



Fixes #30781 and 
Fixes #30780 and 
Part of #30902





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
